packages/core/echo/echo-db/src/client/echo-client.ts

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Context, LifecycleState, Resource } from '@dxos/context';
+import { invariant } from '@dxos/invariant';
+import { PublicKey } from '@dxos/keys';
+import { type DataService } from '@dxos/protocols/proto/dxos/echo/service';
+import { ComplexMap } from '@dxos/util';
+
+import { AutomergeContext } from '../automerge';
+import { EchoDatabaseImpl } from '../database';
+import { Hypergraph } from '../hypergraph';
+
+export type EchoClientParams = {};
+
+export type ConnectToServiceParams = {
+  dataService: DataService;
+};
+
+export type CreateDatabaseParams = {
+  // TODO(dmaretskyi): Change to string id.
```

Comment:
why string?

Reply:
1. space id being a public key is more of a HALO concept
2. I'm not sure if it will stay a PublicKey in HALO - we might wanna move to DIDs

String just seems more flexible, since for ECHO its just an opaque identifier